src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -254,6 +254,40 @@ internal static class UriHelper
             return dest;
         }
 
+        internal static unsafe char[] UnescapeString(string input, int start, int end, char[] dest,
+            ref int destPosition, char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser? syntax,
+            bool isQuery)
+        {
+            fixed (char* pStr = input)
```

Comment:
All this unsafe code in general makes me nervous.  Do we have an issue tracking converting all of this to use spans instead of pointers?  Even if there was a small regression in perf, it'd be worthwhile, and I bet we could actually get it to be faster... in general this whole existing implementation seems from from ideal.

Reply:
I don't think we should be adding new unsafe code. There is already unsafe code in Uri that we want to eventually change to using Span/ReadonlySpan.

Reply:
I'm currently seeing what it would look like to just rewrite it from scratch.  Things have progressed a lot since this code was originally written, and I suspect we can be safer, more concise, and more efficient.  Will see how far I get.